src/txn/txn.c

```diff
@@ -145,7 +145,7 @@ __wt_txn_release_snapshot(WT_SESSION_IMPL *session)
  *     Allocate a snapshot.
  */
 void
-__wt_txn_get_snapshot(WT_SESSION_IMPL *session)
```

Comment:
It's unfortunate that all existing callers need to change, and we've just been burned by passing a locked parameter with the wrong setting to update_serial. I think I'd change this so that this function is `__txn_get_snapshot_int` and there are two others `__wt_txn_get_snapshot` `__wt_txn_get_snapshot_no_publish` or similar.

Reply:
+1 from me.  Maybe consider naming the new one `__wt_txn_bump_snapshot` or similar.